### PR TITLE
[4.1] Fix Debug plugin to dump $_POST requests

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

This enable Debug plugin to collect data while post request (save, cancel tasks, etc).
This was introduced a couple years ago #22327, but broken somwhere in the midle of 4.0 development (due to some refactoring in Application classes).

### Testing Instructions
Apply patch, enable debug.
Try to edit any article, save, cancel etc.
Then open debug bar "history viewer"
image

### Actual result BEFORE applying this Pull Request
In history you can see and select only GET requests.

### Expected result AFTER applying this Pull Request
In history you can see and select GET and POST requests.
image

### Documentation Changes Required
none

### CODE REVIEW

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on c77d011ad96527c638a469ee1a216e4da19ac957<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37465">issues.joomla.org/tracker/joomla-cms/37465</a>.</sub>

File: plugins/system/debug/debug.php [hunk 1]
```diff
@@ -245,7 +266,7 @@ public function onBeforeCompileHead()
 	public function onAfterRespond()
 	{
 		// Do not render if debugging or language debug is not enabled.
-		if (!JDEBUG && !$this->debugLang || $this->isAjax || !($this->app->getDocument() instanceof HtmlDocument))
+		if (!JDEBUG && !$this->debugLang || $this->isAjax)
```
Comment (brianteeman) [inline plugins/system/debug/debug.php:269]: pretty sure this should be reverted to ensure the plugin only processes html

Comment (Fedik) [reply plugins/system/debug/debug.php:269]: Plugin should collect data for all requests (eg format=ajax, format=xml etc), but render only for HTML.
This piece of code bloking data collection for non HTML, that why it removed.

The check for rendering already exists, couple lines down:
https://github.com/joomla/joomla-cms/blob/c77d011ad96527c638a469ee1a216e4da19ac957/plugins/system/debug/debug.php#L327-L329

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on c77d011ad96527c638a469ee1a216e4da19ac957<br /><br />Tested successfully in Joomla 4.1.3-dev of 4 April in Wampserver 3.2.7 using PHP 8.0.15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37465">issues.joomla.org/tracker/joomla-cms/37465</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37465">issues.joomla.org/tracker/joomla-cms/37465</a>.</sub>

File: plugins/system/debug/debug.php [hunk 2]
```diff
@@ -136,6 +138,25 @@ class PlgSystemDebug extends CMSPlugin
...
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public static function getSubscribedEvents(): array
+	{
+		return [
+			'onBeforeCompileHead' => 'onBeforeCompileHead',
+			'onAjaxDebug'         => 'onAjaxDebug',
+			'onBeforeRespond'     => 'onBeforeRespond',
+			'onAfterRespond'      => 'onAfterRespond',
+			ApplicationEvents::AFTER_RESPOND => 'onAfterRespond',
```
Comment (bembelimen) [inline plugins/system/debug/debug.php:155]: Aren't this two lines identical?

Comment (Fedik) [reply plugins/system/debug/debug.php:155]: nope, and that the issue, should be somehow fixed in J5, I guess

Comment (Fedik) [reply plugins/system/debug/debug.php:155]: Application class trigger diferent AfterRespond event for normal request, and for redirect (that usualy happen after POST)

File: plugins/system/debug/debug.php (hunk 1 above)
Comment (Fedik) [reply plugins/system/debug/debug.php:269]: I have changed comment, should be more clear now

File: plugins/system/debug/debug.php (hunk 2 above)
Comment (Fedik) [reply plugins/system/debug/debug.php:155]: Here for `respond`
https://github.com/joomla/joomla-cms/blob/c604191d6f11ee08bf3bfd9edd719e7c492194b5/libraries/src/Application/WebApplication.php#L160-L161
and there for `redirect`
https://github.com/joomla-framework/application/blob/e7b950d2d1358c0baac95a8633a60de20a1e82ab/src/AbstractWebApplication.php#L519-L521

**Comment by chmst** [comment]
Tested after latest changes. It is still RTC and can be merged when tests are ok.

**Comment by laoneo** [comment]
Thanks!!

**Pull request merged**